crates/gpui/src/platform/linux/x11/window.rs

```diff
@@ -268,31 +271,34 @@ impl X11WindowState {
     }
 
     pub fn destroy(&self) {
-        self.inner.lock().renderer.destroy();
+        self.inner.borrow_mut().renderer.destroy();
         self.xcb_connection.send_request(&x::UnmapWindow {
             window: self.x_window,
         });
         self.xcb_connection.send_request(&x::DestroyWindow {
             window: self.x_window,
         });
-        if let Some(fun) = self.callbacks.lock().close.take() {
+        if let Some(fun) = self.callbacks.borrow_mut().close.take() {
             fun();
         }
         self.xcb_connection.flush().unwrap();
     }
 
     pub fn refresh(&self) {
-        let mut cb = self.callbacks.lock();
-        if let Some(ref mut fun) = cb.request_frame {
+        let mut cb = self.callbacks.borrow_mut();
+        if let Some(mut fun) = cb.request_frame.take() {
```

Comment:
what was the motivation for doing this? did we end up calling a callback within a callback?